sympy/geometry/point.py

```diff
@@ -394,19 +506,22 @@ def intersection(self, o):
         [Point2D(0, 0)]
 
         """
-        if isinstance(o, Point):
-            if len(self) != len(o):
+        other = Point.pointify(other, ensure_point=False)
+        if isinstance(other, Point):
+            if len(self) != len(other):
                 raise ValueError("Points must be of the same dimension to intersect")
-            if self == o:
+            if self == other:
                 return [self]
             return []
 
-        return o.intersection(self)
+        return other.intersection(self)
 
     def dot(self, p2):
         """Return dot product of self with another Point."""
-        p2 = Point(p2)
-        return Add(*[a*b for a,b in zip(self, p2)])
+        p2 = Point.pointify(p2, dimension=self.ambient_dimension)
+        # sum could return native python floats, we'd like to ensure the
```

Comment:
If there isn't a test of this, one should be added, e.g., one where `p2 = (0.1, 0.2)`